frontend/public/components/monitoring/alert-manager-config.tsx

```diff
@@ -125,45 +216,127 @@ const RoutingLabel: React.FC<RoutingLabelProps> = ({ labels }) => {
+    history.push('/monitoring/alertmanagerconfig');
+  });
+};
+
+const editReceiverMenuItem = (receiverName: string, canUseEditForm: boolean) => {
+  const menuItem = {
+    label: `Edit ${canUseEditForm ? 'Receiver' : 'YAML'}`,
+  };
+  _.set(menuItem, 'callback', () => {
+    const targetUrl = canUseEditForm
+      ? `/monitoring/alertmanagerconfig/receivers/${receiverName}/edit`
+      : `/monitoring/alertmanageryaml`;
+    return history.push(targetUrl);
+  });
+  return menuItem;
+};
+
+const deleteReceiverMenuItem = (
+  receiverName: string,
+  isDefaultReceiver: boolean,
+  canDelete: boolean,
+) => ({
+  label: 'Delete Receiver',
+  isDisabled: isDefaultReceiver || !canDelete,
+  tooltip:
+    isDefaultReceiver || !canDelete
+      ? 'Cannot delete the default receiver, or a receiver which has a sub-route'
+      : '',
+  callback: () =>
+    confirmModal({
+      title: 'Delete Receiver',
+      message: `Are you sure you want to delete receiver '${receiverName}' ?`,
+      btnText: 'Delete Receiver',
+      executeFn: () => deleteReceiver(receiverName),
+    }),
+});
+const receiverMenuActions = (
+  receiverName: string,
+  isDefaultReceiver: boolean,
+  canUseEditForm: boolean,
+  canDelete: boolean,
+) => [
+  editReceiverMenuItem(receiverName, canUseEditForm),
+  deleteReceiverMenuItem(receiverName, isDefaultReceiver, canDelete),
+];
+const ReceiverKebab = ({ receiverName, isDefaultReceiver, canUseEditForm, canDelete }) => (
+  <Kebab
+    options={receiverMenuActions(receiverName, isDefaultReceiver, canUseEditForm, canDelete)}
+  />
+);
+
 const ReceiverTableRow: React.FC<ReceiverTableRowProps> = ({
   obj: receiver,
   index,
   key,
   style,
-  customData: routingLabels,
+  customData,
 }) => {
+  const { routingLabelsByReceivers, defaultReceiverName } = customData;
   // filter to routing labels belonging to current Receiver
-  const receiverRoutingLabels = _.filter(routingLabels, { receiver: receiver.name });
+  const receiverRoutingLabels = _.filter(routingLabelsByReceivers, { receiver: receiver.name });
+  const integrationTypes = getIntegrationTypes(receiver);
+  const canUseEditForm = isValidForEditForm(receiver, receiverRoutingLabels, integrationTypes);
+  const canDelete = isValidForDelete(receiver, receiverRoutingLabels);
+  const isDefaultReceiver = receiver.name === defaultReceiverName;
   return (
     <TableRow id={index} index={index} trKey={key} style={style}>
       <TableData className={tableColumnClasses[0]}>{receiver.name}</TableData>
-      <TableData className={tableColumnClasses[1]}>{getIntegrationTypes(receiver)}</TableData>
+      <TableData className={tableColumnClasses[1]}>{_.join(integrationTypes, ', ')}</TableData>
       <TableData className={tableColumnClasses[2]}>
-        {_.map(receiverRoutingLabels, (route, i) => (
-          <div key={i}>
-            <RoutingLabel labels={_.get(route, 'labels')} />
+        {isDefaultReceiver && (
+          <div>
```

Comment:
Do we need the wrapping `<div>`?

Reply:
yes and no :-) we do need them to 'group' the routing labels into a single 'label'.  Note each set of routing labels is on a separate line:
![image](https://user-images.githubusercontent.com/12733153/67965384-56d7c600-fbd8-11e9-9d71-a0c623553a60.png)
Without the `<div>`'s, they all run together:
![image](https://user-images.githubusercontent.com/12733153/67965468-75d65800-fbd8-11e9-985b-bf462d720c19.png)

However, the `<div/>`'s do not have be be repeated here, moved them into the `<RoutingLabel  />` component.